### ISSUE TITLE
docs: Improve wording on stating that a method does not work in-place

### ISSUE THREAD

**Issue by zzril** [Safe-DS/Stdlib#346]
### Describe the bug

For methods that return a copy, rather than working in-place, I believe the standard wording in docstrings is:
```py
"""
Returns a new table that/where/with [...].

[...]
"""
```

Currently, we have this sentence in most of our docstrings:
```py
"""
[...]
This table is not modified.
[...]
"""
```
Might be unclear for some users what "this table" refers to.

**This rule should also be added to the guidelines.**

---

Certainly wrong is [this](https://github.com/Safe-DS/Stdlib/blob/main/src/safeds/data/tabular/containers/_table.py#L1126-L1130):
```py
    def tag_columns(self, target_name: str, feature_names: list[str] | None = None) -> TaggedTable:
        """
        Mark the columns of the table as target column or feature columns. The original table is not modified.

        This table is not modified.
        [...]
        """
```

### Expected behavior

Agree on a standard wording to use everywhere.

Definitely fix the redundancy in `tag_columns` in `src/safeds/data/tabular/containers/_table.py`.

**Comment by zzril** [comment]
> **This rule should also be added to the guidelines.**

This part of the issue will be resolved by #427.

**Comment by zzril** [comment]
Blocked by #58 (which will add a bunch of new methods / docstrings in the `TaggedTable` class).
